python/tests/test_tables.py

```diff
@@ -2266,3 +2266,374 @@ def test_set_columns_not_implemented(self):
+            self.assertEqual(pp, p)
+        edges = [
+            i
+            for i, e in enumerate(tables.edges)
+            if e.parent in nodes and e.child in nodes
+        ]
+        self.assertEqual(subset.edges.num_rows, len(edges))
+        for k, e in zip(edges, subset.edges):
+            ee = tables.edges[k]
+            self.assertEqual(ee.left, e.left)
+            self.assertEqual(ee.right, e.right)
+            self.assertEqual(node_map[ee.parent], e.parent)
+            self.assertEqual(node_map[ee.child], e.child)
+            self.assertEqual(ee.metadata, e.metadata)
+        muts = []
+        sites = []
+        for k, m in enumerate(tables.mutations):
+            if m.node in nodes:
+                muts.append(k)
+                if m.site not in sites:
+                    sites.append(m.site)
+        site_map = np.repeat(-1, tables.sites.num_rows)
+        site_map[sites] = np.arange(len(sites), dtype="int32")
+        mutation_map = np.repeat(tskit.NULL, tables.mutations.num_rows + 1)
+        mutation_map[muts] = np.arange(len(muts), dtype="int32")
+        self.assertEqual(subset.sites.num_rows, len(sites))
+        for k, s in zip(sites, subset.sites):
+            ss = tables.sites[k]
+            self.assertEqual(ss, s)
+        self.assertEqual(subset.mutations.num_rows, len(muts))
+        for k, m in zip(muts, subset.mutations):
+            mm = tables.mutations[k]
+            self.assertEqual(mutation_map[mm.parent], m.parent)
+            self.assertEqual(site_map[mm.site], m.site)
+            self.assertEqual(node_map[mm.node], m.node)
+            self.assertEqual(mm.derived_state, m.derived_state)
+            self.assertEqual(mm.metadata, m.metadata)
+        migs = [i for i, mig in enumerate(tables.migrations) if mig.node in nodes]
+        self.assertEqual(subset.migrations.num_rows, len(migs))
+        for k, mig in zip(migs, subset.migrations):
+            mmig = tables.migrations[k]
+            self.assertEqual(mmig.left, mig.left)
+            self.assertEqual(mmig.right, mig.right)
+            self.assertEqual(node_map[mmig.node], mig.node)
+            self.assertEqual(pop_map[mmig.source], mig.source)
+            self.assertEqual(pop_map[mmig.dest], mig.dest)
+            self.assertEqual(mmig.time, mig.time)
+            self.assertEqual(mmig.metadata, mig.metadata)
+        nsp = subset.provenances.num_rows
+        ntp = tables.provenances.num_rows
+        self.assertTrue((nsp == ntp) or (nsp == ntp + 1))
+
+    def test_subset_all(self):
+        # subsetting to everything shouldn't change things
+        ts = self.get_msprime_mig_example(
+            Ne=100, sample_size=10, mutations_per_branch=2
+        )
+        tables = ts.tables
+        tables2 = tables.copy()
+        tables2.subset_nodes(np.arange(ts.num_nodes))
+        # except that you need one extra row in the provenances
+        self.assertEqual(tables.provenances.num_rows + 1, tables2.provenances.num_rows)
```

Comment:
how about do record_prov=False and not need to clear the provenances?

Reply:
it seems the default in other tskit functions is `record_provenance=True` so I wanted to keep it consistent.

Reply:
That should dfeinately be the default! I was saying that in this call to subset, you ask it to not record provenance.